#484 Set * box-sizing to border-box

eduardseifert opened the pull request:
Fix #484

- Added universal star selector for `box-sizing: border-box` (normalization, reset, streamline)

# Test URLs

- Before: https://main--hlx-test--urfuwo.hlx.page/design/design-system
- After: https://484-box-sizing-border-box--hlx-test--urfuwo.hlx.page/design/design-system (looks the same)

alexcarol commented:
The footer looks different:

Main:
![image](https://github.com/urfuwo/hlx-test/assets/1713440/1f8d9da6-2e28-4355-80f6-da0b1dd25b32)
PR:
https://484-box-sizing-border-box--hlx-test--urfuwo.hlx.page/design/design-system

eduardseifert commented:
@alexcarol I repaired all the occurences in the footer. Due to the changes to the box-sizing a few changes were made:

# (1) Contact block:

- moved to nested CSS
- changed padding to margin
- A-Z
- formatting

# (2) Footer block

- moved to nested CSS
- changed padding to margin
- A-Z
- formatting
- small UX enhancements
- kept most of the units

Icons are back in place. The main probs resulted in padding (content-box) vs margin (border-box).

alexcarol commented on aemedge/blocks/contact/contact.css:
```diff
@@ -2,80 +2,82 @@
-  width: 65.4px;
-  padding-right: var(--udexSpacer20);
+.contact-wrapper {
+  .logo-lang-picker {
```
Do we have any policies regarding minimum support for css features? This feature seems to be supported in browsers with ~80% of the users (https://caniuse.com/css-nesting), is this enough for this site? /cc @saurabh-khare @mhaack

eduardseifert commented on aemedge/templates/article/article.js:
```diff
@@ -7,7 +7,7 @@ function decorate(doc) {
   containerize(main, '.hero');
   const tocFlag = getMetadata('toc');
-  if (tocFlag && tocFlag.content !== 'no' && tocFlag.content !== 'false') {
+  if (tocFlag && tocFlag !== 'no' && tocFlag !== 'false') {
```
This came through the rebase. IDK. @rjwtrmn + @saurabh-khare.

alexcarol replied:
I just made a copy of the branch and rebased with main, and I get this diff:
https://github.com/urfuwo/hlx-test/compare/484-box-sizing-border-box-rebase?expand=1
How i did it:
```git fetch origin + git rebase origin/main```

What command did you use to rebase?

eduardseifert commented:
@rjwtrmn + @saurabh-khare Also please have a look on this. Due to the changes in box-sizing I had to update contact block and footer block (used the opportunity to rearange the styles).
